lib/plugins/aws/package/compile/events/apiGateway/lib/deployment.js

```diff
@@ -13,7 +13,6 @@ module.exports = {
         Type: 'AWS::ApiGateway::Deployment',
         Properties: {
           RestApiId: { Ref: this.apiGatewayRestApiLogicalId },
-          StageName: this.options.stage,
```

Comment:
I also create separate stage objects in my alias plugin to be able to have multiple alias deployments within one stage.
From my experience I can say, that the only thing that will happen with this change is, that you probably have to manually delete the stage that was auto-generated with the previous deployment when switching to separate stage resources. (CF will tell you that the stage already exists). But this would only be a one time action.
The other thing is, that the stage must be made dependent on the deployment.
FYI: Having a dedicated stage object is also the only proper way to offer cache configuration and APIG logging configuration for deployed APIs and would be needed anyway in the future.

See my code here: https://github.com/HyperBrain/serverless-aws-alias/blob/master/lib/stackops/apiGateway.js#L77-L94